### DISCUSSION TITLE
UART0 stops working after defining UART2

### DISCUSSION THREAD

**Discussion by Michaeltblaine** [micropython/micropython#11946] [ESP32]
I wrote a program to use UART to talk to a motor driver IC and I have been having trouble using print() function after defining the UART2 connection to the IC. I also tried defining UART0 and writing to it instead of using the print() function. Am I doing anything stupid? Here is the code:

`from machine import UART`

`uart1 = UART(0)`
`uart1.write(b'part1')`
`uart2 = UART(2, baudrate=115200, tx=16, rx=17)`
`uart2.write(b'nothing')`
`uart1.write(b'part2')`

All I receive from UART0 is 'part1'. Part2 is never received.

Thank you in advanced!
my micropython version is: esp32-20220618-v1.19.1.bin

**Comment by robert-hh** [comment]
Pins 16 and 17 might be used for the ESP32 flash. Try other pins for UART 2.

**Reply by Michaeltblaine** [reply]
Thank you so much! I changed the pins and it did work. I don't see any documentation about not using those pins though. How did you figure that? Is it specific to the esp32 pico d4? Thank you!

**Comment by robert-hh** [comment]
You have to look into the schematics of the respective board. But pins 16 and 17 are often used for the memory, and there have been quite a few posts about it. Pins 16 and 17 have initially be used as default pin for UART1. But that caused many hiccups like yours, and so it was changed.

**Reply by Michaeltblaine** [reply]
Well thanks for the help! Saved me some headaches
